eng/pipelines/templates/steps/build-test.yml

```diff
@@ -25,7 +25,7 @@ steps:
 
   - template: eng/pipelines/templates/scripts/verify-agent-os.yml@azure-sdk-tools
     parameters:
-      OSName: ${{ parameters.OSName }}
+      OSName: ${{ parameters.OSVmName }}
```

Comment:
We are now calling this without parameters, assuming that the $(OSVMImage) is already set. see [here](https://github.com/Azure/azure-sdk-for-python/blob/9d242098da666311c76ae82de07bf5161c7929dd/eng/common/pipelines/templates/steps/verify-agent-os.yml#L3) and [here](https://github.com/Azure/azure-sdk-for-python/blob/9d242098da666311c76ae82de07bf5161c7929dd/eng/common/pipelines/templates/steps/verify-agent-os.yml#L18)